drivers/flash/flash_util.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <errno.h>
+
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include <string.h>
+
+LOG_MODULE_REGISTER(flash, CONFIG_FLASH_LOG_LEVEL);
+
+int z_impl_flash_fill(const struct device *dev, uint8_t val, off_t offset,
```

Comment:
Want @henrikbrixandersen's opinion on this, I don't think (and this was said by others in e.g. previous architecture working group discussions) a **driver** should be implementing things that the device itself does not support, that would be a subsystem feature, this is a complete software feature, it has no relation to a hardware feature.

Reply:
My opinion is that driver represents device capabilities, that is all it does. If device does not have something driver should not fix that.
I share that opinion with this comment: https://github.com/zephyrproject-rtos/zephyr/pull/70828#issuecomment-2027497793, fifth paragraph ("Flash drivers must never do hidden erases ...").